src/modules/hip/kernel/resample.hpp

```diff
@@ -0,0 +1,298 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - resample kernel device helpers  --------------------
+
+__device__ __forceinline__ float resample_hip_compute(float &x, float &scale, float &center, float *lookup, int &lookupSize)
+{
+    float locRaw = x * scale + center;
+    int locFloor = std::floor(locRaw);
+    float weight = locRaw - locFloor;
```

Comment:
suggest to combine these so compiler can optimize better
float weight = x * scale + center - std::floor(locRaw);

Reply:
Hi Rajy,
we are storing  x * scale + center in locRaw
and  locFloor = floor(locRaw)

This locFloor is again used in the next steps, which is why we stored x * scale + center and std::floor(locRaw) in separate variables
```
    locFloor = std::max(std::min(locFloor, lookupSize - 2), 0);
    float current = lookup[locFloor];
    float next = lookup[locFloor + 1];
```